lib/home_page.dart

```diff
@@ -127,6 +140,41 @@ class HomeState extends State<Home> with WidgetsBindingObserver {
       });
     });
 
+    commentBloc = Provider.of<NostrCommentBloc>(context, listen: false);
+
+    widget.marketplaceBloc.nostrSettingsStream.listen((event) {
```

Comment:
All these wiring between blocs should not be in the UI layer. More than that it indicates there it too much coupling between these blocs and maybe consider unify them to one NostrBloc. I need to see the NostrCommentBloc before I suggest anything.
If this is not an option we can always pass one bloc stream to another and this stream dependency can happen inside the bloc.

Reply:
Actually, it is to connect the signevent method and getpubkey method of the nostrBloc to the nostrCommentBloc inside at any time. It might not be the best place to do this, but I wanted to be able to fetch both nostrBloc and nostrCommentBloc to be able to connect them.